adam/experiment/observer.py

```diff
@@ -90,6 +90,52 @@ def report(self) -> None:
         )
 
 
+@attrs(slots=True)
+class CandidateAccuracyObserver(
+    DescriptionObserver[SituationT, LinguisticDescriptionT, PerceptionT]
+):
+    r"""
+    Log how often the 'gold' description is present in the learner's candidate descriptions.
+    Provide an accuracy score.
+    """
+    name: str = attrib(validator=instance_of(str))
+    _num_predictions: int = attrib(init=False, default=0)
+    _num_predictions_with_gold_in_candidates: int = attrib(init=False, default=0)
+
+    def observe(  # pylint:disable=unused-argument
+        self,
+        situation: Optional[SituationT],
+        true_description: LinguisticDescription,
+        perceptual_representation: PerceptualRepresentation[PerceptionT],
+        predicted_descriptions: Mapping[LinguisticDescription, float],
+    ) -> None:
+        self._num_predictions += 1
+
+        descriptions_as_token_sequences = [
+            desc.as_token_sequence() for desc in predicted_descriptions
+        ]
+        if true_description.as_token_sequence() in descriptions_as_token_sequences:
+            self._num_predictions_with_gold_in_candidates += 1
+
+    def report(self) -> None:
+        logging.info(
+            f"{self.name}: accuracy of learner's predictions ('gold' description was in learner's candidates) "
```

Comment:
`pylint` will yell at you for using f-strings in logging (because it's inefficient if the log level is below `info`).  Prefer to put `%s`s in the string and then the things to intepolate as additionall arguments to `.info`.